opentreemap/works_management/models.py

```diff
@@ -21,10 +22,28 @@ class Team(models.Model):
 class WorkOrder(Auditable, models.Model):
     instance = models.ForeignKey(Instance)
     name = models.CharField(max_length=255, null=False, blank=False)
+    reference_num = models.IntegerField(default=0)
     created_at = models.DateTimeField(auto_now_add=True)
     created_by = models.ForeignKey(User)
     updated_at = models.DateTimeField(auto_now=True)
 
+    class Meta:
+        unique_together = ('instance', 'reference_num')
+
+    def get_next_reference_num(self):
+        """
+        Return next sequential reference number for instance.
+        """
+        agg_result = Instance.objects.filter(id=self.instance.id) \
+            .annotate(max_value=Max('workorder__reference_num'))[0]
+        max_value = agg_result.max_value or 0
+        return max_value + 1
```

Comment:
I couldn't find any examples of how to use query expressions to create new records with an aggregate value. And I don't think `select_for_update` will work because there's nothing to select until we create the record initially.

This code assumes that objects are created within a transaction. I'm curious to see what happens when multiple workers try to insert tasks at the same time.

Alternatively, we could create 2 new fields on `Instance` to store the "next" values, and use `select_for_update` to safely modify that each time we create a new WorkOrder/Task.